### ISSUE TITLE
Docker image delivers 500 instead of oeplatform

### ISSUE THREAD

**Issue by ebroda** [OpenEnergyPlatform/oeplatform#1790]
## Description of the issue
If I start the docker image using the docker-compose, after fixing the issue from #1787 and building my own image following the instructions on last section [/docker/USAGE.md](https://github.com/OpenEnergyPlatform/oeplatform/blob/develop/docker/USAGE.md#build-the-oeplatform-image), oeplatform is initialized, but the oeplatform is not served afterwards. Instead the browsers shows an apache default error page.

Meanwhile the log of the docker container:
```
oeplatform-1                    | Compressed 11 block(s) from 134 template(s) for 1 context(s).
oeplatform-1                    | Starting apache2
oeplatform-1                    | AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.25.0.3. Set the 'ServerName' directive globally to suppress this message
oeplatform-1                    | AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.25.0.3. Set the 'ServerName' directive globally to suppress this message
oeplatform-1                    | [Fri Aug 16 15:20:17.105520 2024] [mpm_event:notice] [pid 58:tid 58] AH00489: Apache/2.4.61 (Debian) mod_wsgi/4.9.4 Python/3.11 configured -- resuming normal operations
oeplatform-1                    | [Fri Aug 16 15:20:17.105731 2024] [core:notice] [pid 58:tid 58] AH00094: Command line: '/usr/sbin/apache2 -D FOREGROUND'
oeplatform-1                    | [Fri Aug 16 15:20:44.588006 2024] [wsgi:error] [pid 59:tid 64] [remote 172.25.0.1:44958] mod_wsgi (pid=59): Failed to exec Python script file '/app/oeplatform/wsgi.py'.
oeplatform-1                    | [Fri Aug 16 15:20:44.588071 2024] [wsgi:error] [pid 59:tid 64] [remote 172.25.0.1:44958] mod_wsgi (pid=59): Exception occurred processing WSGI script '/app/oeplatform/wsgi.py'.
oeplatform-1                    | [Fri Aug 16 15:20:44.613163 2024] [wsgi:error] [pid 59:tid 64] [remote 172.25.0.1:44958] Traceback (most recent call last):
oeplatform-1                    | [Fri Aug 16 15:20:44.617862 2024] [wsgi:error] [pid 59:tid 64] [remote 172.25.0.1:44958]   File "/app/oeplatform/wsgi.py", line 12, in <module>
oeplatform-1                    | [Fri Aug 16 15:20:44.617882 2024] [wsgi:error] [pid 59:tid 64] [remote 172.25.0.1:44958]     from django.core.wsgi import get_wsgi_application
oeplatform-1                    | [Fri Aug 16 15:20:44.617902 2024] [wsgi:error] [pid 59:tid 64] [remote 172.25.0.1:44958] ModuleNotFoundError: No module named 'django'
oeplatform-1                    | 172.25.0.1 - - [16/Aug/2024:15:20:44 +0000] "GET / HTTP/1.1" 500 608
```
It seems like there's some problem with the dependencies. But connecting to the container and running pip install -r requirements.txt says that all dependencies are installed. 

What's interesting is the debug line:
```
[...] Apache/2.4.61 (Debian) mod_wsgi/4.9.4 Python/3.11 configured [...]
```
Checking the container, there's a python3.11 executable (v3.11) and an python3, which is v3.10.
So somehow mod_wsgi is using a different version (3.11) as the normal image (3.10) - so mod_wsgi does not know about the packages installed for 3.10.

## Steps to Reproduce
1. Build image according to changes in #1787
2. Start docker image
3. Open browse and access localhost:8080

## Ideas of solution
First idea was if it's possible to change the python path. But this stackoverflow post](https://stackoverflow.com/a/40951638) indicates that mod_wsgi is hardly connected to the python version. Therefore, my try was to increase the Python version in the Dockerfile (FROM python:3.11.9) to hope that in that case it's the same python version being used for mod_wsgi as also for the container itself. That now works.

And I didn't notice problems somewhere with the app, but only tested shortly. Do you have any known issues in regard to Python 3.11?

## Workflow checklist
- [x] I am aware of the workflow in [CONTRIBUTING.md](https://github.com/OpenEnergyPlatform/oeplatform/blob/develop/CONTRIBUTING.md)

**Comment by jh-RLI** [comment]
We are currently using Python 3.10 in production. We finally switched from Python 3.6 a few months ago and also had to update mod_wsgi for the new Python version. The docker image doesn't just have to reflect our infrastructure setup, but that was the idea so far (and after your finding: is not the case currently).

As we are still using django 3.2, only python 3.10 is supported in the docs.
https://docs.djangoproject.com/en/5.0/faq/install/#what-python-version-can-i-use-with-django

We should build mod_wsgi for each image build with the correct python version, I assume. But I'd have to take a closer look to make more meaningful comments here.

**Comment by ebroda** [comment]
Okay, that's an argument against 3.11. 

Maybe this description helps to get the running again:
https://github.com/GrahamDumpleton/mod_wsgi/issues/467#issuecomment-533934942

I played a bit around and first tried it with a venv, as suggested in the issue:
```
RUN python -m venv /py-venv/
ENV PATH="/py-venv/bin:$PATH"

RUN pip install -r requirements.txt
RUN pip install mod_wsgi
RUN mod_wsgi-express module-config >> /etc/apache2/apache2.conf
```
which worked. But I thought, if that's really necessary and remove the venv again, as that's not necessary as there's only the mod_wsgi now that I've installed via pip. So it seems to be enough to have only the pip install of mod_wsgi.

I tried it out in #1802 and this leads to a correct version of mod_wsgi being used there.
> oeplatform-1 [...] Apache/2.4.61 (Debian) mod_wsgi/5.0.0 Python/3.10 configured -- resuming normal operations
